package.json

```diff
@@ -39,47 +26,46 @@
     "jshint": "^2.9.5",
     "minimist": "^1.2.0",
     "mocha": "^3.4.2",
-    "moustache": "0.0.2",
     "npm-run-all": "^4.1.1",
     "nsp": "^2.8.1",
-    "onchange": "^3.2.1",
     "prettier": "^1.10.2",
     "selenium-webdriver": "^3.5.0",
-    "shield-studies-addon-utils": "^4.1.0"
+    "web-ext": "^2.4.0"
   },
   "engines": {
     "node": ">=8.9.0"
   },
   "homepage": "http://github.com/mozilla/shield-studies-addon-template",
   "keywords": [
     "firefox",
-    "legacy-addon",
     "mozilla",
-    "shield-study"
+    "shield-study",
+    "template"
   ],
-  "license": "MIT",
-  "main": "index.js",
+  "license": "MPL-2",
   "repository": {
     "type": "git",
     "url": "git://github.com/mozilla/shield-studies-addon-template.git"
   },
   "scripts": {
-    "build": "bash ./bin/xpi.sh",
+    "build": "web-ext build",
+    "bundle-utils": "cp node_modules/shield-studies-addon-utils/dist/StudyUtils.jsm src/privileged/shieldUtils/jsm/StudyUtils.jsm",
     "eslint": "eslint . --ext jsm --ext js --ext json",
     "eslint-fix": "npm run eslint -- --fix",
-    "firefox": "export XPI=dist/linked-addon.xpi && npm run build && node run-firefox.js",
     "format": "prettier '**/*.{css,js,json,jsm,md}' --trailing-comma=all --ignore-path=.eslintignore --write",
-    "postformat": "npm run eslint-fix",
-    "harness_test": "export XPI=dist/linked-addon.xpi && mocha test/functional_tests.js --retry 2 --reporter json",
+    "harness_test": "export ADDON_ZIP=./dist/button_icon_preference_study_shield_study_example_-2.0.0.zip && mocha test/functional_tests.js --retry 2 --reporter json",
     "lint": "npm-run-all lint:*",
-    "lint-build:addons-linter": "# actually a post build test:  bin/addonLintTest ' + require('./package.json').name",
-    "lint:addons-linter": "addons-linter addon/webextension/",
+    "lint:addons-linter": "addons-linter src/",
     "lint:eslint": "npm run eslint",
     "lint:fixpack": "fixpack",
     "lint:nsp": "nsp check",
-    "prebuild": "cp node_modules/shield-studies-addon-utils/dist/StudyUtils.jsm addon/",
+    "lint:web-ext-lint": "web-ext lint",
+    "postformat": "npm run eslint-fix",
+    "prebuild": "npm run bundle-utils",
+    "prewatch": "npm run bundle-utils",
     "sign": "echo 'TBD, see: https://bugzilla.mozilla.org/show_bug.cgi?id=1407757'",
-    "test": "export XPI=dist/linked-addon.xpi && npm run build && mocha test/functional_tests.js --retry 2",
-    "watch": "onchange 'addon/**' 'package.json' 'template/**' -e addon/install.rdf -e addon/chrome.manifest -e addon/StudyUtils.jsm -- npm run build -- '{{event}} {{changed}} $(date)'"
+    "start": "web-ext run --no-reload",
+    "test": "export ADDON_ZIP=./dist/button_icon_preference_study_shield_study_example_-2.0.0.zip && npm run build && mocha test/functional_tests.js --retry 2",
```

Comment:
How would this work once the version number changes, since it looks like we're hard coding "_-2.0.0.zip" in the path? Or is this file not generated?

Reply:
This needs to be addressed, but the way to issue this depends on https://github.com/mozilla/web-ext/issues/855#issuecomment-375214750